interfaces/Portalicious/src/app/components/query-table/query-table.component.ts

```diff
@@ -484,9 +489,43 @@ export class QueryTableComponent<TData extends { id: PropertyKey }, TContext> {
    */
   visibleColumns = model<QueryTableColumn<TData>[]>([]);
 
-  resetColumnVisibility() {
+  private getStoredColumns(stateKey: string): null | QueryTableColumn<TData>[] {
+    const storedColumns = localStorage.getItem(stateKey);
+    if (!storedColumns) return null;
+
+    return JSON.parse(storedColumns) as QueryTableColumn<TData>[];
+  }
+
+  private getMatchingColumns(
+    storedColumns: QueryTableColumn<TData>[],
+  ): QueryTableColumn<TData>[] {
+    return storedColumns
+      .map((column) => this.columns().find((c) => c.field === column.field))
+      .filter(Boolean) as QueryTableColumn<TData>[];
```

Comment:
```suggestion
.filter((column) => !!this.columns().find((c) => c.field === column.field));
```

The double loop made its way back again :D

Also I don't think the ` as ` is necessary here.

Reply:
Ahhh okok but then maybe?

```suggestion
      .map((column) => this.columns().find((c) => c.field === column.field))
      .filter(column => column !== undefined)
```